### PR TITLE
followers/following count synchronize on follow/unfollow buttons click

### PR DESCRIPTION
**Pull request by Micheduc25**
- Number of followers and number of followings from the ProfileView and from the UserProfile Card are now synchronized, and are reactive.

### CODE REVIEW

File: eru/src/store/modules/authentication/actions.js
Comment (viryl15) [inline eru/src/store/modules/authentication/actions.js:140]: @Micheduc25 this method exist before your commit right?
I'm asking this because the whole method is mark as a new bloc of code, it's not normal

File: eru/src/views/User/ProfileView.vue
Comment (viryl15) [inline eru/src/views/User/ProfileView.vue:22]: @Micheduc25 I can see that you changed the variable name here, why?
@FotieMConstant I asked you to change that variable name before to something like "data or result"....

File: eru/src/components/modules/cards/createScreelCard.vue
Comment (viryl15) [inline eru/src/components/modules/cards/createScreelCard.vue:138]: is this line correct?

**Review by viryl15 — CHANGES_REQUESTED** [review]

File: eru/src/store/modules/authentication/actions.js
Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places

File: eru/src/views/User/ProfileView.vue
Comment (Micheduc25) [reply eru/src/views/User/ProfileView.vue:22]: I used a name which was more descriptive. "theUser" is kind of vague.

Comment (Micheduc25) [reply eru/src/views/User/ProfileView.vue:22]: I created a userInView object in the store, this object keeps the current user whose profile is being viewed. I did this so that when you click on the follow button the number of followers on the ProfileViewCard should be updated

File: eru/src/components/modules/cards/createScreelCard.vue
Comment (Micheduc25) [reply eru/src/components/modules/cards/createScreelCard.vue:138]: Yes, the code which was there before was throwing an error in the console because the "document.getElementById('create-screel-btn')" was undefined at times. so I first of all check if the element exists before toggling the "shake-button" class

File: eru/src/store/modules/authentication/actions.js
Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places

There was not existing action for that in the store?

Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> 
> There was not existing action for that in the store?
Yes there was no action for that

Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: The only actions which were there were actions to login using different providers

Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > 
> > 
> > There was not existing action for that in the store?
> > Yes there was no action for that

Have you not seen the getSpecificUserAction mothod in user module in the store?

Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > 
> > > 
> > > There was not existing action for that in the store?
> > > Yes there was no action for that
> 
> Have you not seen the getSpecificUserAction mothod in user module in the store?

Yes i saw it, but it does not set the currentUser object in the state

Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > 
> > > > 
> > > > There was not existing action for that in the store?
> > > > Yes there was no action for that
> > 
> > 
> > Have you not seen the getSpecificUserAction mothod in user module in the store?
> 
> Yes i saw it, but it does not set the currentUserObject in the state

please, why not just do it (set the currentUserObject in the state) in place of creating another method?

Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: > > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > > 
> > > > > 
> > > > > There was not existing action for that in the store?
> > > > > Yes there was no action for that
> > > 
> > > 
> > > Have you not seen the getSpecificUserAction mothod in user module in the store?
> > 
> > 
> > Yes i saw it, but it does not set the currentUserObject in the state
> 
> please, why not just do it (set the currentUserObject in the state) in place of creating another method?

ok let me try that

File: eru/src/views/User/ProfileView.vue
Comment (viryl15) [reply eru/src/views/User/ProfileView.vue:22]: > I used a name which was more descriptive. "theUser" is kind of vague.

Maybe but Next time, please don't change existing variable name like that, so for this time it's ok

File: eru/src/store/modules/authentication/actions.js
Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: But i will have to do some modifications to the getSpecificUserAction to be able to use it in the router/index.js file. I can't access the Vue _vm from there

Comment (FotieMConstant) [reply eru/src/store/modules/authentication/actions.js:140]: A way you could go about this is to pass the _vm down as prop to the function like i did in some functions in the store

Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: > But i will have to do some modifications to the getSpecificUserAction to be able to use it in the router/index.js file. I can't access the Vue _vm from there

why can you not access  the Vue _vm? I am seeing something like payload._vm inside the method

File: eru/src/views/User/ProfileView.vue
Comment (Micheduc25) [reply eru/src/views/User/ProfileView.vue:22]: > > I used a name which was more descriptive. "theUser" is kind of vague.
> 
> Maybe but Next time, please don't change existing variable name like that, so for this time it's ok

ok noted

File: eru/src/store/modules/authentication/actions.js
Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: > I way you could go around this is to pass the _vm down as prop go the function link i did in some functions in the store

Yes, what I am saying is that the getSpecificUserAction requires the _vm as an argument, but i need to call that action from the router.js file and there is no way for me to access the current _vm from the router/index.js file.

Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: > > I way you could go around this is to pass the _vm down as prop go the function link i did in some functions in the store
> 
> Yes, what I am saying is that the getSpecificUserAction requires the _vm as an argument, but i need to call that action from the router.js file and there is no way for me to access the current _vm from the router/index.js file.

ok nice, your new method has a purpose

**Review by viryl15 — APPROVED** [review]

File: eru/src/store/modules/authentication/actions.js
Comment (FotieMConstant) [reply eru/src/store/modules/authentication/actions.js:140]: Why does he need to call the function in the router again? Don't really get that point though

Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: > > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > > 
> > > > > 
> > > > > There was not existing action for that in the store?
> > > > > Yes there was no action for that
> > > 
> > > 
> > > Have you not seen the getSpecificUserAction mothod in user module in the store?
> > 
> > 
> > Yes i saw it, but it does not set the currentUserObject in the state
> 
> please, why not just do it (set the currentUserObject in the state) in place of creating another method?

And another advantage of the action i created is that we can get the current user using just the token.
- I am facing difficulties using the getSpecificUserAction, because it requires the _vm and the username, but I don't have access to all of that in the router/index.js file

Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: > > > > > > > Yes it is an action i created to be able to get the current user's data from the backend. so that we that need rewrite the same code to get the user from the backend in several places
> > > > > > 
> > > > > > 
> > > > > > There was not existing action for that in the store?
> > > > > > Yes there was no action for that
> > > > 
> > > > 
> > > > Have you not seen the getSpecificUserAction mothod in user module in the store?
> > > 
> > > 
> > > Yes i saw it, but it does not set the currentUserObject in the state
> > 
> > 
> > please, why not just do it (set the currentUserObject in the state) in place of creating another method?
> 
> And another advantage of the action i created is that we can get the current user using just the token.
> 
> * I am facing difficulties using the getSpecificUserAction, because it requires the _vm and the username, but I don't have access to all of that in the router/index.js file

Ok, I think it's ok, I'll approve this PR
So let me test it before

Comment (Micheduc25) [reply eru/src/store/modules/authentication/actions.js:140]: > Why does he need to call the function in the router again? Don't really get that point though

So that we get the most recent value for the user object each time the site is reloaded

Comment (FotieMConstant) [reply eru/src/store/modules/authentication/actions.js:140]: Oh yeah, @viryl15 i think he's right 👍🏾

Comment (FotieMConstant) [reply eru/src/store/modules/authentication/actions.js:140]: I haven't reviewed his code in detail...  But i believe he makes a point... @viryl15 i trust you'll review it good and approve once done... I'm not home at the moment

Comment (viryl15) [reply eru/src/store/modules/authentication/actions.js:140]: I tested and it works

**Pull request merged**
